### ISSUE TITLE
(Auth) AuthEvents Question

### ISSUE THREAD

**Issue by mshaffer12882** [myflashlab/Firebase-ANE#46]
Is there a better way to parse though AuthEvents msg?

For example in the case Auth.RESULT_AUTH_USER_COLLISION:
e.msg could be:
ERROR_EMAIL_ALREADY_IN_USE
ERROR_ACCOUNT_EXISTS_WITH_DIFFERENT_CREDENTIAL
ERROR_CREDENTIAL_ALREADY_IN_USE

is there a better way to check the error_name besides parsing e.msg?

**Comment by Soshaapp** [comment]
I do it like this:

Auth.listener.addEventListener( AuthEvents.CREATE_NEW_USER_RESULT, createNewUserResult );
Auth.createUserWithEmailAndPassword( email, password );

protected function createNewUserResult( e : AuthEvents ) : void
{
     Auth.listener.removeEventListener( AuthEvents.CREATE_NEW_USER_RESULT, createNewUserResult );
     if( e.result == Auth.RESULT_AUTH_INVALID_CREDENTIALS )
     {
     }
     else if( e.result == Auth.RESULT_AUTH_USER_COLLISION )
     {
     }
}

Hope it helps :)

/Martin

**Comment by mshaffer12882** [comment]
@Soshaapp Thanks for the help.  Maybe I should be more clear on my question.  I understand how to use the basic listener calls.  But take RESULT_AUTH_USER_COLLISION for example it has sub errors so you can tell exactly what went wrong.  Some of the sub errors are:
ERROR_EMAIL_ALREADY_IN_USE 
ERROR_ACCOUNT_EXISTS_WITH_DIFFERENT_CREDENTIAL
ERROR_CREDENTIAL_ALREADY_IN_USE 

How it works right now AuthEvents.msg stores the full error message from firebase, which I have to parse though to get the error.  I have now built a class that I can call that will return the "sub error", however it feels like that should be built into the ane.

**Comment by myflashlab** [comment]
@mshaffer12882 Totally understand your question. While we were building the ANE, we thought of that in fact but we finally decided to keep things like how it is now (similar to the native side). Adding seperated listeners would have added too much unnecessary confusion to the ANE and too keep things identical between the iOS and the Android side, we had to take this approach. Thanks for bringing that up anyhow. I will share this information with the team for future reviews.
